---
title: "Tasks"
format: 
  html: 
    toc: true
    toc-depth: 2
    toc-location: right
    toc-expand: false
    toc-fold: true
    number-sections: false
    toc-title: Contents
    smooth-scroll: true
    lightbox: true

---

Please do the following tasks always after the corresponding session. We will look at your results and problems that occurred always in the beginning of the next session.

At the end of the course make a protocol of all your splicing analyses in form of a quarto or rmarkdown report (.qmd/.rmd) and render it to pdf or html. Include all tasks below in the protocol.

\newline
\newline
\newline
\newline
\newline
\newline
\newline
\newline



# Tasks for part 1 - Visualisation

Instructions for screening AS events in the VOILA GUI and verification via Sashimi Plots in IGV can be found in *part1_GUI_and_SashimiPlots.pdf*.

1.  For one of your comparisons search for binary LSVs with |deltaPSI| > 0.1 and Confidence Threshold > 0.9. If you can't find events lower the deltaPSI threshold. The screenshots should include the Gene Name, the LSV ID, the LSV Schema and the violin plot of the dPSI values.
  +   Take a screenshot of 1 LSV representing an exon skipping event.
  +   Take a screenshot of 1 LSV representing an intron retention event.
  +   Take a screenshot of 1 LSV representing an alternative 5' splice site event.
  +   Take a screenshot of 1 LSV representing an alternative 3' splice site event.
2.  Take for each of the five LSVs also screenshots for your other data set (e.g. your other cell line). You can use the LSV identifier to find them. If you can't find the same events lower the deltaPSI and confidence threshold. You can lower both thresholds to 0 to see all detected LSVs. If you still cannot find them use take screenshots of other events.
3.  Compare the regulation of the five LSVs between the two comparisons.
4.  Take one of your five LSVs and create a sashimi plot including all of your comparisons in IGV.
  +   You need to load the BAM-Files into IGV and navigate to the region where the LSV is located. It is sufficient if you use the BAM-File of one replicate per condition.
  +   If the BAM-Files are too huge for your computer subset them in Galaxy to the chromosome on which the LSV is located.
5. Does the sashimi plot confirm what you have seen in the VOILA GUI?


# Tasks for part 2 - Analyzing AS at the level of LSVs

+ Repeat or revisit the functions in the"useful R" section. (This part does not have to be included in the proctocol in the end.)

+ Analyse both of you data sets following the steps of the "Global analysis" section. 

+ How many LSVs are detected in each data set? How many are significantly regulated? 

+ What are the top regulated LSVs in each data set?

+ Do the data sets have a similar depth? (Are the numbers of detected and regukated LSVS similar?)

+ What is the most common splicing class for your protein? Do a small literature research on the protein, that was knocked down / knocked out in you data set. Which types of splicing would you expect? Is this also what the data shows?

# Tasks for part 3 - Comparison of both data sets

Compare the splicing in both your data sets using steps from the "Comparison of data sets" section.

+ Do the data sets give comparable results? If not, what could be the reasons?

# Tasks for part 4 - Functional context of alternative splicing events

+ Have a look at your alternative gene expression analysis. Which of the genes from the top 20 LSVs are also regulated alternatively? Are there any? Which type of splicing is happening inthese genes?

+ Identify two interesting alternative splicing events for which vastDB provides functional annotations (available publications).
Can you see some effect on one or several cancer types?
